provider/kubernetes/kubernetes.go

```diff
@@ -299,6 +322,16 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 								break
 							}
 
+							if percentageWeight, found := backendPercentageWeightMap[pa.Backend.ServiceName]; found {
+								instanceCount := 0
+								for _, subset := range endpoints.Subsets {
+									instanceCount = instanceCount + len(subset.Addresses)
+								}
+								serverWeight = int(percentageWeight.RawValue()) / instanceCount
+							} else if leftFractionPercentage < 1 {
+								serverWeight = int(PercentageValueFromFloat64(leftFractionPercentage).RawValue() / int64(leftFractionInstanceCount))
```

Comment:
instead of:
```go
	serverWeight = int(percentageWeight.RawValue()) / instanceCount
} else if leftFractionPercentage < 1 {
	serverWeight = int(PercentageValueFromFloat64(leftFractionPercentage).RawValue() / int64(leftFractionInstanceCount))
```
I think you can add a method `computeWeight/getWeight` on the struct `PercentageValue`:

```go
func (p *PercentageValue) computeWeight(count int64) int { ... }
```